### PR TITLE
Create interface and abstract classes for scene manager

### PR DESCRIPTION
**Pull request by Thomaltarix**
Since the scene manager was duplicated in the client and in the server, I created an interface and an abstract class for it.

Obviously, I updated both sides according to this new system.

### CODE REVIEW

File: SceneManager/ISceneManager.hpp
Comment (Peralban) [inline SceneManager/ISceneManager.hpp:1]: No Empty line :(

Comment (Thomaltarix) [reply SceneManager/ISceneManager.hpp:1]: Done

File: SceneManager/ASceneManager.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                    if (stringKeyMap.find(key.asString()) == stringKeyMap.end())
+                        throw SceneManagerJsonErrors("Error while loading the key: " + key.asString());
+                    _keysSystems[index][stringKeyMap.at(key.asString())] = system;
+                } else {
+                    throw SceneManagerJsonErrors("Error while loading the key: " + key.asString());
+                }
+            }
+        } else {
+            throw SceneManagerJsonErrors("Error while loading the key: " + key.asString());
+        }
```
Comment (Peralban) [inline SceneManager/ASceneManager.cpp:113]: Awful, but alright.

**Review by Peralban — APPROVED** [review]
Maybe you can transform your code in a more beautiful one.

File: SceneManager/ASceneManager.cpp (hunk 1 above)
Comment (Thomaltarix) [reply SceneManager/ASceneManager.cpp:113]: You're right, I'll change it right away

File: SceneManager/ASceneManager.cpp
Comment (Peralban) [inline SceneManager/ASceneManager.cpp:1]: It's more readable like that, good job :+1:

**Review by Peralban — APPROVED** [review]

**Review by mathieurobert1 — APPROVED** [review]
Perfect, thank you for this huge change. This is clearer.

**Pull request merged**